api/v1alpha1/openapi.yaml

```diff
@@ -1622,19 +1622,54 @@ components:
         - metadata
         - spec
       description: Repository represents a git repository
-    RepositorySpec:
+    RepositoryHttpConfig:
       type: object
       properties:
-        repo:
-          type: string
-          description: 'The (possibly remote) repository URL to clone from'
         username:
           type: string
           description: 'The username for auth with HTTP transport'
         password:
           type: string
           description: 'The password for auth with HTTP transport'
           format: password
+        tls.crt:
+          type: string
+          description: 'Base64 encoded TLS cert data'
+          format: password
+        tls.key:
+          type: string
+          description: 'Base64 encoded TLS cert key'
+          format: password
+        ca.crt:
+          type: string
+          description: 'Base64 encoded root CA'
+        skipServerVerification:
+          type: boolean
+          description: 'Skip remote server verification'
+    RepositoryGitConfig:
+      type: object
+      properties:
+        sshPrivateKey:
+          type: string
+          description: 'The private SSH key'
+          format: password
+        privateKeyPassphrase:
+          type: string
+          description: 'The passphrase for sshPrivateKey'
+          format: password
+        skipServerVerification:
+          type: boolean
+          description: 'Skip remote server verification'
+    RepositorySpec:
+      type: object
+      properties:
+        repo:
```

Comment:
Let's move this into both GitHttpConfig and GitSSHConfig

Reply:
I was wondering what we would put here for a k8s secret?  The name?  Seems weird.

Reply:
It would be similar to the git https repo, probably the connection details for the k8s service account and endpoint.